windows-gaming-driver/src/driver/control/codec.rs

```diff
@@ -50,3 +52,40 @@ impl Encoder for Codec {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bytes::BytesMut;
+    use tokio_io::codec::Decoder;
+
+    macro_rules! test {
+        (fn $name:ident, with $data:expr, expect $expected:expr, len $len:expr) => (
```

Comment:
If you don't want to understand the `test!` macro, it's easier to read and understand the test cases, which is the reason I added them.